### PR TITLE
Add new parameters features

### PR DESCRIPTION
**Pull request by sicking**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -178,13 +181,15 @@ declare global {
...
 
   interface SuggestionResults {
-    setSuggestions: (suggestions: Array<string | { name: string; data?: any }>) => void
+    setSuggestions(suggestions: Array<string | { name: string; data?: any; icon?: (string | Uint8Array); iconUrl?: string }>): void
+    setError(message: string): void
+    setLoadingMessage(message: string): void
   }
 
-  type ParameterInputEvent = {
+  type ParameterInputEvent<ParametersType = ParameterValues> = {
```
Comment (johndoherty) [inline index.d.ts:189]: What does this do?

**Review by johndoherty — APPROVED** [review]
Looks good, just a question about the `ParameterType` generic

**Comment by johndoherty** [comment]
Might be worth checking with the plugin samples to make sure they all still compile.

File: index.d.ts (hunk 1 above)
Comment (sicking) [reply index.d.ts:189]: This is to enable people to write

```
interface MyParams {
  node: { id: string }
  color: string
  height: string
}
figma.parameters.on('input', ({ parameters, query, key, result } : ParameterInputEvent<MyParams>) => {
  ...
})

I.e. you can specify your own type to use for the `parameters` object. But you can also use the previous syntax and get the same result as before

Comment (johndoherty) [reply index.d.ts:189]: Ah cool!

**Pull request merged**